src/hints/mod.rs

```diff
@@ -1,140 +1,173 @@
+            }
+            code => Err(HintError::UnknownHint(code.to_string().into_boxed_str())),
+        }
+    }
+}
 
-    let check_is_deprecated_hint = HintFunc(Box::new(check_is_deprecated));
-    hint_processor.add_hint(String::from(CHECK_IS_DEPRECATED), Rc::new(check_is_deprecated_hint));
+pub struct SnosSimpleHintProcessor {
+    builtin_hint_proc: BuiltinHintProcessor,
+    run_resources: RunResources,
+}
 
-    let is_deprecated_hint = HintFunc(Box::new(is_deprecated));
-    hint_processor.add_hint(String::from(IS_DEPRECATED), Rc::new(is_deprecated_hint));
+impl ResourceTracker for SnosSimpleHintProcessor {
+    fn consumed(&self) -> bool {
+        self.run_resources.consumed()
+    }
 
-    let os_context_segments_hint = HintFunc(Box::new(os_context_segments));
-    hint_processor.add_hint(String::from(OS_CONTEXT_SEGMENTS), Rc::new(os_context_segments_hint));
+    fn consume_step(&mut self) {
+        self.run_resources.consume_step()
+    }
 
-    let selected_builtins_hint = HintFunc(Box::new(selected_builtins));
-    hint_processor.add_hint(String::from(SELECTED_BUILTINS), Rc::new(selected_builtins_hint));
+    fn get_n_steps(&self) -> Option<usize> {
+        self.run_resources.get_n_steps()
+    }
 
-    let select_builtin_hint = HintFunc(Box::new(select_builtin));
-    hint_processor.add_hint(String::from(SELECT_BUILTIN), Rc::new(select_builtin_hint));
+    fn run_resources(&self) -> &RunResources {
+        &self.run_resources
+    }
+}
 
-    let enter_call_hint = HintFunc(Box::new(enter_call));
-    hint_processor.add_hint(String::from(ENTER_CALL), Rc::new(enter_call_hint));
+impl Default for SnosSimpleHintProcessor {
+    fn default() -> Self {
+        Self { builtin_hint_proc: BuiltinHintProcessor::new_empty(), run_resources: Default::default() }
+    }
+}
 
-    hint_processor
+impl HintProcessorLogic for SnosSimpleHintProcessor {
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn core::any::Any>,
+        constants: &HashMap<String, Felt252>,
+    ) -> Result<(), HintError> {
+        match self.builtin_hint_proc.execute_hint(vm, exec_scopes, hint_data, constants) {
+            Err(HintError::UnknownHint(_)) => {}
+            res => return res,
+        }
+        let hint_data = hint_data.downcast_ref::<HintProcessorData>().ok_or(HintError::WrongHintData)?;
+
+        let hint_func = match &*hint_data.code {
+            STARKNET_OS_INPUT => starknet_os_input,
+            INITIALIZE_STATE_CHANGES => initialize_state_changes,
+            INITIALIZE_CLASS_HASHES => initialize_class_hashes,
+            SEGMENTS_ADD => segments_add,
+            SEGMENTS_ADD_TEMP => segments_add_temp,
+            TRANSACTIONS_LEN => transactions_len,
+            block_context::LOAD_CLASS_FACTS => block_context::load_class_facts,
```

Comment:
What is the reason for not importing: `block_context::LOAD_CLASS_FACTS` it would be: `LOAD_CLASS_FACTS => load_class_facts,`.

Reply:
This way the hint processor becomes self documenting. The variable names themselves are good, but the mod prefix is explicit. 

Not tied to it, if you feel strongly to remove.